### PR TITLE
COMP: Remove unused variables

### PR DESCRIPTION
**Pull request by hjmjohnson**
Variables that are never read can be removed.

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)

### CODE REVIEW

**Comment by N-Dekker** [comment]
@hjmjohnson  Thanks Hans! How did you find these cases? I mean, did they trigger a specific code analysis message or compiler warning?

File: Modules/Filtering/Smoothing/include/itkBinomialBlurImageFilter.hxx [hunk 1]
```diff
@@ -146,15 +146,15 @@ BinomialBlurImageFilter<TInputImage, TOutputImage>::GenerateData()
   typename TTempImage::IndexType indexShift;
 
   // How many times has the algorithm executed? (for debug)
-  int num_reps = 0;
+  itkDebugStatement(int num_reps = 0);
```
Comment (N-Dekker) [inline Modules/Filtering/Smoothing/include/itkBinomialBlurImageFilter.hxx:149]: Thanks Hans, but why not just removing this `num_reps` variable? Is the BinomialBlurImageFilter still actively being debugged, after so many years? (Looks like the filter was added to ITK more than 20 years ago: commit d236c7aa0ee7c361b2920c09ef9d170c583d5de2, Damion Shelton, June 7, 2001)

File: Modules/Filtering/Smoothing/include/itkBinomialBlurImageFilter.hxx [hunk 2]
```diff
@@ -146,15 +146,15 @@ BinomialBlurImageFilter<TInputImage, TOutputImage>::GenerateData()
...
 
   // Temporary pixel storage
   double pixelA, pixelB;
 
   // walk the output image forwards and compute blur
   for (unsigned int rep = 0; rep < m_Repetitions; ++rep)
   {
-    ++num_reps;
+    itkDebugStatement(++num_reps);
```
Comment (N-Dekker) [inline Modules/Filtering/Smoothing/include/itkBinomialBlurImageFilter.hxx:157]: Is `num_reps` always equal to `rep`?

File: Modules/Filtering/Smoothing/include/itkBinomialBlurImageFilter.hxx [hunk 3]
```diff
@@ -145,17 +145,12 @@ BinomialBlurImageFilter<TInputImage, TOutputImage>::GenerateData()
...
-
   // Temporary pixel storage
   double pixelA, pixelB;
 
   // walk the output image forwards and compute blur
   for (unsigned int rep = 0; rep < m_Repetitions; ++rep)
   {
-    ++num_reps;
-
     itkDebugMacro(<< "Repetition #" << rep);
```
[the remark this answers is not in the document]
Comment (hjmjohnson) [reply Modules/Filtering/Smoothing/include/itkBinomialBlurImageFilter.hxx:154]: Good idea, but that is out of scope for this PR.  I'm just trying to clear warnings that started showing up for recent clang compilations on my computer.

[the remark this answers is not in the document]
Comment (jhlegarreta) [reply Modules/Filtering/Smoothing/include/itkBinomialBlurImageFilter.hxx:154]: I understand Hans; not sure why https://github.com/InsightSoftwareConsortium/ITK/pull/3996/files#diff-d71e1482d96a19d53d807b667a2f429c97bf4d4cbb4d5abbaa33d2f93968bbc8L241 would trigger a warning, but OK. Maybe stating them in the commit message would be helpful.

[the remark this answers is not in the document]
Comment (dzenanz) [reply Modules/Filtering/Smoothing/include/itkBinomialBlurImageFilter.hxx:154]: `itkDebugMacro` only does something in debug mode. So if compiled in release mode, a warning appears with Hans' compiler.

**Review by N-Dekker — APPROVED** [review]
Thanks Hans! Can you still tell us how you spotted these cases? Do they appear somewhere on the CI? Or did you locally run some specific code analysis? Anyway, approved 👍

**Comment by dzenanz** [comment]
@N-Dekker You missed this comment: https://github.com/InsightSoftwareConsortium/ITK/pull/3996#discussion_r1158720736. Some new version of clang produces the warning.

**Comment by hjmjohnson** [comment]
> Thanks Hans! Can you still tell us how you spotted these cases? Do they appear somewhere on the CI? Or did you locally run some specific code analysis? Anyway, approved 👍

I have a new M2 Mac computer, and these error showed up during a "Release" build. 

```
➜  ~ clang++ --version 
Apple clang version 14.0.3 (clang-1403.0.22.14.1)
Target: arm64-apple-darwin22.4.0
Thread model: posix
InstalledDir: /Library/Developer/CommandLineTools/usr/bin
```

**Pull request merged**